overlord/snapstate/snapstate.go

```diff
@@ -2016,10 +2049,20 @@ func autoRefreshPhase1(ctx context.Context, st *state.State) ([]string, []*state
 
 	// return all names as potentially getting updated even though some may be
 	// held.
-	names := make([]string, len(updates))
+	var names []string
 	toUpdate := make(map[string]*refreshCandidate, len(updates))
-	for i, up := range updates {
-		names[i] = up.InstanceName()
+	for _, up := range updates {
+		// if specific gating snap was requested, filter out updates.
+		if forGatingSnap != "" {
+			affecting, ok := affectedSnaps[forGatingSnap]
+			if !ok {
+				continue
+			}
+			if _, ok := affecting.AffectingSnaps[up.InstanceName()]; !ok {
```

Comment:
it's bool and again it seems we coud read this much earlier,  we are basically finding the intersection between updated & forGatingSnap affecting snaps, also again we should ignore affectingSnaps if the update is actually forGatingSnap